### ISSUE TITLE
Shared Element Transition used in Android having a side effect for the app on iOS

### ISSUE THREAD

**Issue by minuhariharan** [wix/react-native-navigation#2231]
### Issue Description
 We are using the Shared Element Transition feature for our app - similar to the react native navigator example for Masonry - where the user clicks on a thumbnail of the profile image, and the enlarged image is shown on the next screen centered. This works fine for Android. I know that SET is not supported for iOS yet, but the image is no longer centered. The image seems to be displaced just for iOS. We have tested this feature without SET and it works fine for iOS. 

### Steps to Reproduce / Code Snippets / Screenshots
Code Snippet: 
```javascript
<View style={styles.screenContainer}>
        <SharedElementTransition
          sharedElementId={this.props.uid}
          showDuration={SHOW_DURATION}
          hideDuration={HIDE_DURATION}
          showInterpolation={{
            type: 'linear',
            easing: 'FastInSlowOut',
          }}
          hideInterpolation={{
            type: 'linear',
            easing: 'FastOutSlowIn',
          }}
        >   
          <Image source={avatarSource}
        style={{width, height}}
        resizeMode={Image.resizeMode.contain} />
        </SharedElementTransition>
      </View>
```

Screenshot after SET in iOS : Image is not centered

<img width="305" alt="screen shot 2017-11-27 at 2 55 04 pm" src="https://user-images.githubusercontent.com/29897898/33259456-f6ce0b54-d382-11e7-9935-7c59ebd3a7a2.png">

Screenshot without SET in iOS: Image is centered

<img width="307" alt="screen shot 2017-11-27 at 2 58 17 pm" src="https://user-images.githubusercontent.com/29897898/33259763-eb55c464-d383-11e7-957d-94fff7800a71.png">

Our actual intent is to use SET for Android and it is working fine there:

<img width="233" alt="screen shot 2017-11-27 at 3 05 59 pm" src="https://user-images.githubusercontent.com/29897898/33259961-7ca46772-d384-11e7-94df-78f686000bdc.png">
### Environment
* React Native Navigation version: 1.1.253
* React Native version: 0.49.3
* Platform(s) (iOS, Android, or both?): Both
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator

**Comment by pqkluan** [comment]
You could work around this issue by using platform-specific code.

Since the `SharedElementTransition` is Android only feature I think the fact this issue happening is natural and don't need to be addressed. I'll close this issue for now, feel free to re-open this if needed.

**Comment by varungupta85** [comment]
The problem was that the `SharedElementTransition` is replaced by a `View` for iOS. So, we need to add a `style` prop to the `SharedElementTransition` that we would want to be applied to the `View` in case of iOS.

**Comment by pqkluan** [comment]
So you have a problem with the platform-specific code.

Create a component variable and props:
```jsx
const Component = Platform.select({
  ios: View,
  android: SharedElementTransition,
});
const ComponentProps = Platform.select({
  ios: {
    // View props goes here
  },
  android: {
    // SharedElementTransition props go here
  },
});

and render part:

```jsx
<Component {...componentProps}>
  // Child component
</Component>
```

Hope this help.

**Comment by varungupta85** [comment]
Yes, we have already fixed the component. I posted the comment for the benefit of others if someone else runs into a similar problem. The thing to note is that `SharedElementTransition` component is just replaced by a `View` in case of iOS. So, if there is some weird layout happening on iOS after adding `SharedElementTransition`, good way to test is to replace the `SharedElementTransition` with just a view, fix the layout and apply the resulting `style` property to the `SharedElementTransition` component itself.
